### PR TITLE
feat(pack-up): add build command

### PR DESCRIPTION
**Pull request by joshuaellis**
### What does it do?

* introduces pack-up package (name can be discussed)
* adds the build command (largely just replicates the plugin:build work)

### Why is it needed?

* so we can build projects without the weight of `@strapi/strapi` deps

### How to test it?

* Automated tests incl. CLI testing 👀 

### Related issues

* resolves CONTENT-1878

### CODE REVIEW

File: packages/utils/pack-up/src/node/core/__tests__/pkg.test.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
+import fs from 'fs/promises';
+import path from 'path';
+
+import { loadPkg, validatePkg } from '../pkg';
+import { Logger } from '../logger';
```
Comment (innerdvations) [inline packages/utils/pack-up/src/node/core/__tests__/pkg.test.ts:5]:
```suggestion
```

File: packages/utils/pack-up/src/node/init.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { CommonCLIOptions } from '../types';
+import { createLogger } from './core/logger';
+
+export interface InitOptions extends CommonCLIOptions {}
```
Comment (innerdvations) [inline packages/utils/pack-up/src/node/init.ts:4]:
```suggestion
export interface InitOptions = CommonCLIOptions;
```
Make lint happy

File: packages/utils/pack-up/src/node/init.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+export const init = async (opts: InitOptions) => {
+  const { silent, debug } = opts;
+
+  const logger = createLogger({ silent, debug });
+
+  console.log('success!');
```
Comment (innerdvations) [inline packages/utils/pack-up/src/node/init.ts:11]: Same; should console.log be allowed with lint?

File: packages/utils/pack-up/src/node/check.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { CommonCLIOptions } from '../types';
+import { createLogger } from './core/logger';
+
+export interface CheckOptions extends CommonCLIOptions {}
```
Comment (innerdvations) [inline packages/utils/pack-up/src/node/check.ts:4]:
```suggestion
export interface CheckOptions = CommonCLIOptions;
```

File: packages/utils/pack-up/src/node/watch.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+export interface WatchOptions extends CommonCLIOptions {
+  cwd?: string;
+}
+
+export const watch = async (opts: WatchOptions) => {
+  const { silent, debug, cwd = process.cwd() } = opts;
+
+  const logger = createLogger({ silent, debug });
+
+  console.log('success!');
```
Comment (innerdvations) [inline packages/utils/pack-up/src/node/watch.ts:13]: same

File: packages/utils/pack-up/src/node/core/config.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+interface ConfigBundle {
+  source: string;
+  import?: string;
+  require?: string;
+  runtime?: Runtime;
+}
+
+interface ConfigOptions {
```
Comment (innerdvations) [inline packages/utils/pack-up/src/node/core/config.ts:65]:
```suggestion
export interface ConfigOptions {
```

File: packages/utils/pack-up/scripts/check.ts [hunk 7]
```diff
@@ -0,0 +1,11 @@
+import path from 'path';
+
+import { check } from '../src/node';
```
[the remark this answers is not in the document]
Comment (joshuaellis) [reply packages/utils/pack-up/scripts/check.ts:3]: I should import from `src` this is solved :)

**Comment by joshuaellis** [comment]
rebased to solve conflicts

**Pull request merged**